TesSense.py

```diff
@@ -33,142 +37,208 @@ def printerror(error, err) :                          # Error message with trunc
+    rate = minrate = 0  # Minimum rate you want to set the charger to
+    volts = 120  # Minimum volts, until detected by the charger
+    pshown = loop = charge = charging = False  # init more variables
+
+    with teslapy.Tesla( username ) as tesla:
+        if not tesla.authorized:
+            tesla.refresh_token(refresh_token=input('Enter SSO refresh token: '))
+
+        vehicles = tesla.vehicle_list()
+        car = vehicles[0].get_vehicle_summary()
+        print(car['display_name'], "is", car['state'], "\n")
+
+        #print( "Tesla Info:\n", vehicles[0].get_vehicle_data(), "\n\n" ) # shows all car data available
+        #print( "TeslaPy:\n", dir( teslapy ), "\n\n" )         # shows all the TeslaPy API functions
+        #print( "Senseable:\n", dir( Senseable ), "\n\n" )     # shows all the Sense API functions
+
+        while (True):
             try :
-                chargedata=vehicles[0].get_vehicle_data()['charge_state']
-            except teslapy.HTTPError as e:
-                printerror("Tesla failed to update, please wait a minute...", e)
-                sleep( 60 )
-                continue
-                
-            if chargedata['charging_state'] == "Disconnected": # Loop w/o msgs until connected
-                if not pshown :
-                    printmsg(" Please plug the vehicle in...\n\n")
-                    pshown = True
-                sleep( 60 )
-                continue
-            else : pshown = False
-            
-            if chargedata['battery_level'] >= chargedata['charge_limit_soc'] : # Loop when full
-                exit("Full Battery!")
-
-            if chargedata['fast_charger_present']:    # Loop while DC Fast Charging
-                printmsg("Supercharging...")
+                sense.update_realtime()                   # Update Sense info
+                asp = int(sense.active_solar_power)
+                ap = int(sense.active_power)
+                power_diff = asp-ap                       # Watts being set back to the grid
+            except :
+                printmsg("Sense Timeout")
+                await asyncio.sleep(10)
+                continue # back to the top of the order
+
+            # Assume not charging to start
+            charging = False
+
+            if vehicles[0].available() :
+                try :
+                    chargedata=vehicles[0].get_vehicle_data()['charge_state']
+                except teslapy.HTTPError as e:
+                    printerror("Tesla failed to update, please wait a minute...", e)
+                    await asyncio.sleep(60)
+                    continue
+
+                if chargedata['charging_state'] == "Disconnected": # Loop w/o msgs until connected
+                    if not pshown :
+                        printmsg(" Please plug the vehicle in...\n\n")
+                        pshown = True
+
+                    # Set 0 watts to plug
+                    mutable_plug.data_source.power = 0
```

Comment:
I picked a few spots where it seemed like you'd want to set the plug power to zero, but correct me if I'm wrong!